model/search_params.go

```diff
@@ -75,56 +75,77 @@ func parseSearchFlags(input []string) ([]string, map[string]string) {
 	return words, flags
 }
 
-func ParseSearchParams(text string) (*SearchParams, *SearchParams) {
+func ParseSearchParams(text string) []*SearchParams {
 	words, flags := parseSearchFlags(splitWords(text))
 
-	hashtagTerms := []string{}
-	plainTerms := []string{}
+	hashtagTermList := []string{}
+	plainTermList := []string{}
 
 	for _, word := range words {
 		if validHashtag.MatchString(word) {
-			hashtagTerms = append(hashtagTerms, word)
+			hashtagTermList = append(hashtagTermList, word)
 		} else {
-			plainTerms = append(plainTerms, word)
+			plainTermList = append(plainTermList, word)
 		}
 	}
 
-	inChannel := flags["channel"]
-	if inChannel == "" {
-		inChannel = flags["in"]
-	}
+	hashtagTerms := strings.Join(hashtagTermList, " ")
+	plainTerms := strings.Join(plainTermList, " ")
+
+	inChannels := []string{}
+	fromUsers := []string{}
 
-	fromUser := flags["from"]
+	for _, flagPair := range flags {
+		flag := flagPair[0]
+		value := flagPair[1]
 
-	var plainParams *SearchParams
-	if len(plainTerms) > 0 {
-		plainParams = &SearchParams{
-			Terms:     strings.Join(plainTerms, " "),
-			IsHashtag: false,
-			InChannel: inChannel,
-			FromUser:  fromUser,
+		if flag == "in" || flag == "channel" {
+			inChannels = append(inChannels, value)
+		} else if flag == "from" {
+			fromUsers = append(fromUsers, value)
 		}
 	}
 
-	var hashtagParams *SearchParams
-	if len(hashtagTerms) > 0 {
-		hashtagParams = &SearchParams{
-			Terms:     strings.Join(hashtagTerms, " "),
-			IsHashtag: true,
-			InChannel: inChannel,
-			FromUser:  fromUser,
-		}
+	if len(inChannels) == 0 {
+		inChannels = append(inChannels, "")
+	}
+	if len(fromUsers) == 0 {
+		fromUsers = append(fromUsers, "")
 	}
 
-	// special case for when no terms are specified but we still have a filter
-	if plainParams == nil && hashtagParams == nil && (inChannel != "" || fromUser != "") {
-		plainParams = &SearchParams{
-			Terms:     "",
-			IsHashtag: false,
-			InChannel: inChannel,
-			FromUser:  fromUser,
+	paramsList := []*SearchParams{}
+
+	for _, inChannel := range inChannels {
```

Comment:
Yeah that actually makes a lot more sense. I just saw how the plain term search vs the hashtag term search happened originally and it never occurred to me to just change the sql query to have something like `IF channel.Name IN ...` in it. I'll make that change.